pepper-apis/src/main/java/org/broadinstitute/ddp/customexport/export/CustomExportCoordinator.java

```diff
@@ -0,0 +1,346 @@
+                    customDto.getGuid(), res.getStatusCode());
+        }
+
+        return true;
+    }
+
+    private Mail createNotificationEmail(boolean isFailure, boolean isSkip) {
+        Config cfg = exportCfg.getConfig(CustomExportConfigFile.EMAIL);
+        String subject;
+        Email fromEmail = new Email(cfg.getString(CustomExportConfigFile.EMAIL_FROM_EMAIL),
+                cfg.getString(CustomExportConfigFile.EMAIL_FROM_NAME));
+        Email toEmail = new Email(cfg.getString(CustomExportConfigFile.EMAIL_TO_EMAIL),
+                cfg.getString(CustomExportConfigFile.EMAIL_TO_NAME));
+
+        List<Content> content = new ArrayList<>();
+        String fileUrl = "https://console.cloud.google.com/storage/browser/" + exportCfg.getString(CustomExportConfigFile.BUCKET_NAME)
+                + "/" + exportCfg.getString(CustomExportConfigFile.FILE_PATH);
+
+        if (isFailure) {
+            subject = cfg.getString(CustomExportConfigFile.EMAIL_ERROR_SUBJECT);
+            content.add(new Content("text/plain", cfg.getString(CustomExportConfigFile.EMAIL_ERROR_CONTENT)));
+        } else if (isSkip) {
+            subject = cfg.getString(CustomExportConfigFile.EMAIL_SKIP_SUBJECT);
+            content.add(new Content("text/plain", cfg.getString(CustomExportConfigFile.EMAIL_SKIP_CONTENT)));
+        } else {
+            subject = cfg.getString(CustomExportConfigFile.EMAIL_SUCCESS_SUBJECT);
+            addSuccessContent(content, cfg.getString(CustomExportConfigFile.EMAIL_SUCCESS_CONTENT), fileUrl);
+        }
+
+        return createMail(subject, fromEmail, toEmail, content, fileUrl);
+    }
+
+    private void addSuccessContent(List<Content> content, String fullContent, String fileUrl) {
+        Content linkContent = new Content("text/html", "<a href=\"" + fileUrl + "\">" + fileUrl + "</a>");
+        if (!fullContent.contains("{{bucketLink}}")) {
+            content.add(new Content("text/plain", fullContent));
+            content.add(linkContent);
+        } else {
+            int startLinkIndex = fullContent.indexOf("{{bucketLink}}");
+            int endLinkIndex = startLinkIndex + "{{bucketLink}}".length();
+            if (startLinkIndex > 0) {
+                content.add(new Content("text/plain", fullContent.substring(0, startLinkIndex)));
+            }
+            content.add(linkContent);
+            if (endLinkIndex < fullContent.length()) {
+                content.add(new Content("text/plain", fullContent.substring(endLinkIndex)));
+            }
+        }
+    }
+
+    private Mail createMail(String subject, Email fromEmail, Email toEmail, List<Content> content, String fileUrl) {
+        Mail mail = new Mail();
+        mail.setFrom(fromEmail);
+        mail.setSubject(subject);
+        Personalization p = new Personalization();
+        p.addTo(toEmail);
+        mail.addPersonalization(p);
+        p.addSubstitution("{{bucketLink}}", "<a href=\"" + fileUrl + "\">" + fileUrl + "</a>");
```

Comment:
`addSubstitution` are for the legacy template syntax (e.g. `-ddp.baseWebUrl-`). I think what you're looking for is `addDynamicTemplateData`. Also, it'll be good to cleanup the `addSuccessContent()` if we're going with dynamic templating approach.